enhancements/user-workload-monitoring.md

```diff
@@ -0,0 +1,275 @@
+in the `openshift-user-monitoring` namespace only.
+It ignores Prometheus custom resources in any other namespace.
+The same holds true for the cluster monitoring prometheus operator (red).
+
+#### Aggregation
+
+The front facing Kubernetes service has to make it as simple as possible
+for integrators to leverage both user workload monitoring as well as cluster monitoring.
+
+This is achieved by using Thanos Querier which allows to aggregate
+and optionally deduplicate the cluster and user workload monitoring metrics backends
+under single Prometheus Query endpoint.
+
+Usage of the Thanos querier component inside cluster monitoring has many advantages:
+1. Aggregation of multiple sources (user workload monitoring and cluster monitoring) into one global view.
+2. Run-time deduplication of metrics available today.
+3. Gap-filling of metrics series in the Prometheus HA user and cluster monitoring setup.
+4. Potentially add the possibility of adding a query cache for increased performance in the future.
+
+See [5] and [6] for more details.
+
+### Tenancy
+
+![](user-monitoring-request.png)
+
+Tenancy is achieved by leveraging the existing topology that is protecting cluster monitoring Prometheus already today.
+The kube-rbac-proxy sidecar will be deployed along with prom-label-proxy in the Thanos Querier Deployment.
+
+For details about the tenancy model see the README for prom-label-proxy [2].
+
+### Risks and Mitigations
+
+Risks are for one in the area of security, namely in the realm of accessing metrics a user has no permissions for.
+This is achieved by reusing the existing tenancy stack as well as with the newly added features for allow-lists,
+deny-lists and reconcilation limiting settings for prometheus operator.
+
+Another risks are related to saturation of Prometheus servers.
+Potential queries of death (QoD) or aggressive scrapte targets with high cardinality
+can impact the availability of the user workload monitoring prometheus instances.
+
+## Design Details
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+N/A
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+There is no dev preview planned, just tech preview.
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Document relationship to user workload monitoring
+when launching prometheus operator via OLM.
+- Specify deprecation strategy for OperatorGroup custom resources in OLM.
+The OperatorGroup is marked as deprecated,
+the removal is blocked by OLM's prometheus operator.
+- Analyze saturation of new prometheus servers
+and evaluate fitness for GA by analyzing telemetry.
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
```

Comment:
Could you address the future of the Prometheus operator in OperatorHub as it relates to this Enhancement?  Will it be deprecated in favor of this?

Reply:
Ideally yes, although I see the final deprecation decision not part of this proposal. We do have a blocker today, as `OperatorGroup` support is deprecated in OLM and its final removal is blocked by monitoring currently, so removal from OLM is the envisioned decision.